source/auth/auth.rst

```diff
@@ -1272,151 +1215,167 @@ mechanism_properties
         callbacks.  This value MUST not be allowed in the URI connection
         string.  The hostname check MUST be performed after SRV record
         resolution, if applicable.
+        This property is only applicable when ``REQUEST_TOKEN_CALLBACK`` is given.
 
-Drivers MUST NOT send a PrincipalStepRequest when performing automatic authentication
-or when there is a cached IdPServerResponse.  Drivers must instead  use ``saslStart`` with a JwtStepRequest.
-
-Speculative Authentication
-```````````````````````````````````
-Drivers MUST implement speculative authentication for MONGODB-OIDC during the ``hello`` handshake.  If there is an unexpired access token, the JwtStepRequest SASL command will be used as the speculation command.  If there is no cache value, the PrincipalStepRequest will be used as the speculation command.  The driver MUST NOT call any callbacks during speculative authentication.
+Interfaces
+``````````
 
-User Provided Callbacks
-```````````````````````
+Authenticating using the MONGODB-OIDC mechanism will require 1 or 2 round trips between the MongoDB driver and server.
+The requests from the driver and the replies from the server are described by the following interfaces which are encoded
+in the payload as octet sequences defining BSON objects:
 
-Drivers MUST allow the user to provide callbacks for token request and
-token refresh.  The driver MUST provide a way for the both callbacks to be either automatically
-canceled, or to cancel itself.  This can be as a timeout argument to the
-callback, a cancellation context passed to the callback, or some other
-language-appropriate mechanism.  The timeout duration MUST be 5 minutes,
-to account for the fact that there may be human interaction involved.
+.. code:: typescript
 
-Callbacks can be synchronous and/or asynchronous, depending on the driver
-and/or language.  Asynchronous callbacks should be preferred when other
-operations in the driver use asynchronous functions.
+  // Driver’s opening request in saslStart.
+  interface PrincipalStepRequest {
+    // Name of the OIDC user Principal.
+    n?: str;
```

Comment:
Optional: The convention for optional values in most other TypeScript code samples in the specifications repo is
```
n: Optional<string>
````
While I don't think that's typical TypeScript syntax, it is a little clearer for people who aren't familiar with TypeScript. Consider using the more conventional (for the specifications repo) `Optional<>` syntax. This suggestion applies to all uses of the `?` operator in this PR.

Reply:
Updated